app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionViewModel.kt

```diff
@@ -45,17 +46,19 @@ class SelectionInteractionViewModel private constructor(
   private val minAllowableSelectionCount: Int by lazy {
     interaction.customizationArgsMap["minAllowableSelectionCount"]?.signedInt ?: 1
   }
-  private val maxAllowableSelectionCount: Int by lazy {
+  val maxAllowableSelectionCount: Int by lazy {
     // Assume that at least 1 answer always needs to be submitted, and that the max can't be less than the min for cases
     // when either of the counts are not specified.
     interaction.customizationArgsMap["maxAllowableSelectionCount"]?.signedInt
       ?: minAllowableSelectionCount
   }
-  private val selectedItems: MutableList<Int> = mutableListOf()
+  val selectedItems: MutableList<Int> = mutableListOf()
   val choiceItems: ObservableList<SelectionInteractionContentViewModel> =
     computeChoiceItems(choiceSubtitledHtmls, hasConversationView, this)
 
   private val isAnswerAvailable = ObservableField(false)
+  val selectedItemText = ObservableField("Please select all correct choices")
```

Comment:
This should use a string ID rather than a hardcoded, in-text string.

Reply:
Okay @BenHenning but I am facing issues in the 3 more where the `maxAllowableSelectionCount ` is used, I am not getting how we can use this in `strings.xml`.

Reply:
You can pass variables into strings by adding a '%s' to the string, and then calling a format method to pass the value into the string. For a reference, see:

https://github.com/oppia/oppia-android/blob/798f08b881f1cc84556d0c30dbb6a1b79b4bbb75/app/src/main/res/values/strings.xml#L17

and

https://github.com/oppia/oppia-android/blob/091b45a1885562752f6d25da8e5494e4a331bdb4/app/src/main/java/org/oppia/android/app/player/audio/AudioViewModel.kt#L149

Reply:
@BenHenning I am getting an error in `resourceHandler` and I have also tried to google it also it's saying that I have to take context but that is also not working in my case.

Reply:
@Akshatkamboj14 I need more information to be able to help:
- What specific error did you run into?
- What code is causing it?
- What have you tried when attempting to fix it?
- What do you think the problem might be?
- How would you proceed next to solve it?
-

Reply:
@BenHenning in the `getString` method I guess I have to get the `context` reference to call the `getString` method and I have tried `this.getString()`, `writtenTranslationContext.getString`, `context.getString()`, `Context.getString()`  and [https://github.com/oppia/oppia-android/pull/4636#discussion_r1007630463](url) the one you have mentioned in the above comments, but they all are not working and I am getting this error,

![image](https://user-images.githubusercontent.com/92685493/198831551-09c12429-25ab-4916-ac1a-33ecfe80f7e9.png)

Reply:
@Akshatkamboj14 have you looked at other view models to see how they are translated? You should also provide more specifics than "doesn't work" as that doesn't provide much context--is it always a symbol error?

Reply:
Hey @BenHenning I have looked and most of them are using `resourceHandler` as their context but in the `SelectionInteractionViewModel` there is no parameter of `resourceHandler `.

![image](https://user-images.githubusercontent.com/92685493/200135621-0f9696a5-7c97-4a6b-b3be-b9bab6611e0c.png)

![image](https://user-images.githubusercontent.com/92685493/200135643-3f349ef4-5d7c-4fc0-b91c-159700450e02.png)

![image](https://user-images.githubusercontent.com/92685493/200135652-f20be027-6856-4bd7-9a31-edf40b6083d1.png)